### PR TITLE
Add support for Scipy v1.14

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
SciPy v1.14 packages `libopenblas` as `libscipy_openblas` on Linux and OpenBlas symbols are now exposed with a prefix `scipy_`.

**Description of the Change:**
Update `UtilLinearAlg.hpp` to check the library name before getting openblas symbols.

**Benefits:**
Support `scipy` v0.14

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Install dependencies
-        run: sudo apt update && sudo apt -y install python3 python3-pip && python -m pip install pip~=22.0 && python3 -m pip install -r requirements-dev.txt
+        run: sudo apt update && sudo apt -y install python3 python3-pip && python -m pip install pip~=24.0 && python3 -m pip install -r requirements-dev.txt
```
Comment (maliasadi) [inline .github/workflows/format.yml:41]:
```suggestion
        run: sudo apt update && sudo apt -y install python3 python3-pip && python -m pip install pip~=22.0 && python3 -m pip install -r requirements-dev.txt
```

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-pip~=22.0
+pip~=24.0
```
Comment (maliasadi) [inline requirements-dev.txt:1]:
```suggestion
pip~=22.0
```

File: pennylane_lightning/core/src/utils/config.h [hunk 3]
```diff
@@ -17,7 +17,8 @@
  * Config file for the path to scipy.libs at compile time.
  */
 
+
 #ifndef CONFIG_H
 #define CONFIG_H
-#define SCIPY_LIBS_PATH ""
+#define SCIPY_LIBS_PATH "/home/ali/miniforge3/envs/cat/lib/python3.11/site-packages/scipy.libs"
```
Comment (maliasadi) [inline pennylane_lightning/core/src/utils/config.h:23]:
```suggestion
#define SCIPY_LIBS_PATH ""
```

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -134,6 +134,9 @@
 * Remove `pennylane.ops.op_math.controlled_decompositions.ctrl_decomp_zyz` tests with `len(control_wires) > 1`.
   [(#757)](https://github.com/PennyLaneAI/pennylane-lightning/pull/757)
 
+* Add support for Scipy v1.14.
+  [(#776)](https://github.com/PennyLaneAI/pennylane-lightning/pull/776)
```
Comment (mlxd) [inline .github/CHANGELOG.md:138]: I assume we still test against <1.14 too in the CI, so that these changes still work with older versions?

**Review by mlxd — COMMENTED** [review]
Just one q:

File: .github/CHANGELOG.md (hunk 4 above)
Comment (vincentmr) [reply .github/CHANGELOG.md:138]: Seems like the CI pulled 1.13.

Comment (maliasadi) [reply .github/CHANGELOG.md:138]: yes! we use < 1.13 in Catalyst and the comp checks pass against this PR.

**Review by mlxd — APPROVED** [review]

**Review by vincentmr — APPROVED** [review]
LGTM, thanks @maliasadi .

**Pull request merged**